app.py

```diff
@@ -10,10 +10,77 @@
 
 app = Flask(__name__)
 
-
 app.config['SEND_FILE_MAX_AGE_DEFAULT'] = 300
 
-db_path = 'student_intern_data/student_intern_data.db'  # Replace with your SQLite database file path
+ # Replace with your SQLite database file path
+db_path = 'student_intern_data/student_intern_data.db' 
+
+# Allocating students projects
+@app.route('/assigned_projects/', methods=['GET', 'PUT'])
+def assigned_projects():
+    # Connect to the SQLite database
+    conn = sqlite3.connect('student_intern_data/student_intern_data.db')
+    cursor = conn.cursor()
+    try:
+        if request.method == 'GET':
+            # Fetch the projects from the Projects table
+            cursor.execute('SELECT id, name FROM Projects ORDER BY id DESC')
+            projects = cursor.fetchall()
+
+            # Fetch the students from the Students table
+            cursor.execute('SELECT intern_id, full_name, project FROM Students')
```

Comment:
As per comment - only need to get students from this intake (see Current Recruitment /current_intake)

Reply:
Done